automation/pg_si_test/pg_si_test.go

```diff
@@ -0,0 +1,280 @@
+	suite.clientset = clientset
+	suite.setupPath = setupPaths
+	suite.config = config
+
+	log.Printf("******************** END PostgresqlSingleInstanceTestSuite SETUPSUITE() ********************\n")
+}
+
+// TearDownSuite is called once after running the tests in the suite
+func (suite *PostgresqlSingleInstanceTestSuite) TearDownSuite() {
+	log.Printf("******************** RUNNING PostgresqlSingleInstanceTestSuite TEARDOWNSUITE() ********************\n")
+
+	var err error
+	setupPaths := suite.setupPath
+
+	dbSecret := &v1.Secret{}
+	err = automation.CreateTypeFromPath(dbSecret, setupPaths.DbSecretPath)
+	if err != nil {
+		log.Printf("Error occurred while executing utils.createSecret() for dbSecret with path %s, error: %v\n", setupPaths.DbSecretPath, err)
+	}
+
+	ndbSecret := &v1.Secret{}
+	err = automation.CreateTypeFromPath(ndbSecret, setupPaths.NdbSecretPath)
+	if err != nil {
+		log.Printf("Error occurred while executing utils.createSecret() for ndbSecret with path %s, error: %v\n", setupPaths.NdbSecretPath, err)
+		suite.T().FailNow()
+	}
+
+	database := &ndbv1alpha1.Database{}
+	err = automation.CreateTypeFromPath(database, setupPaths.DbPath)
+	if err != nil {
+		log.Printf("Error occurred while executing utils.createSecret() for database with path %s, error: %v\n", setupPaths.DbPath, err)
+		suite.T().FailNow()
+	}
+
+	appPod := &v1.Pod{}
+	err = automation.CreateTypeFromPath(appPod, setupPaths.AppPodPath)
+	if err != nil {
+		log.Printf("Error occurred while executing utils.createPod() for pod with path %s, error: %v\n", setupPaths.AppPodPath, err)
+		suite.T().FailNow()
+	}
+
+	appSvc := &v1.Service{}
+	err = automation.CreateTypeFromPath(appSvc, setupPaths.AppSvcPath)
+	if err != nil {
+		log.Printf("Error occurred while executing utils.createService() for pod with path %s, error: %v\n", setupPaths.AppSvcPath, err)
+		suite.T().FailNow()
+	}
+
+	err = automation.TestTeardown(dbSecret, ndbSecret, database, appPod, appSvc, suite.clientset, suite.v1alpha1ClientSet, suite.T())
+	if err != nil {
+		log.Printf("Error occurred: %s\n", err)
+		log.Println("teardown failed")
+	} else {
+		log.Println("teardown completed")
+	}
+
+	suite.v1alpha1ClientSet.Databases(database.Namespace)
+
+	log.Printf("******************** END PostgresqlSingleInstanceTestSuite() ********************\n")
+}
+
+// This will run right before the test starts and receives the suite and test names as input
+func (suite *PostgresqlSingleInstanceTestSuite) BeforeTest(suiteName, testName string) {
+	log.Printf("******************** RUNNING TEST %s %s ********************\n", suiteName, testName)
+}
+
+// This will run after test finishes and receives the suite and test names as input
+func (suite *PostgresqlSingleInstanceTestSuite) AfterTest(suiteName, testName string) {
+	log.Printf("******************** END TEST %s %s ********************\n", suiteName, testName)
+}
+
+func (suite *PostgresqlSingleInstanceTestSuite) TestProvisioningSuccess() {
```

Comment:
This function looks a little cluttered. Please add break lines. Also add comments.